src/io.rs

```diff
@@ -209,4 +209,82 @@ mod tests {
             assert!(hasher.validate(&mut rd).is_ok() == *valid);
         }
     }
+
+    #[test]
+    fn test_copy_n() {
```

Comment:
If this PR  does not progress quick enough, this commit can be split to its own PR and merged separately.